src/main/java/com/cloudbees/jenkins/plugins/bitbucket/server/client/BitbucketServerAPIClient.java

```diff
@@ -1197,4 +1197,31 @@ public InputStream getFileContent(BitbucketSCMFile file) throws IOException, Int
         }
         return content;
     }
+
+    @Restricted(ProtectedExternally.class)
+    protected CloseableHttpResponse executeMethod(HttpRequestBase httpMethod)
+            throws IOException {
+        CloseableHttpClient client = getHttpClient(httpMethod);
+        CloseableHttpResponse response = client.execute(httpMethod, context);
+        while (response.getStatusLine().getStatusCode() == TOO_MANY_REQUESTS_HTTP_STATUS) {
```

Comment:
Maybe good to put a limit here like 300 times (~25min)

Reply:
Agree, handled.